### PR TITLE
feat: Add comprehensive KNN join integration tests and benchmarks

### PR DESCRIPTION
**Pull request by zhangfengcdt**
**Summary**
- Add integration tests for KNN join functionality with synthetic data
- Include cross-verification against PostGIS for correctness validation
- Add comprehensive benchmarking comparing SedonaDB, PostGIS, and DuckDB
- Test various scenarios: basic joins, polygon joins, edge cases, and attribute preservation
- Performance results show SedonaDB is 8-655× faster than competitors

**Integration Tests Added**
  - **Basic KNN Join**: Point-to-point KNN queries with configurable k values
  - **Mixed Geometry Types**: Point-to-polygon KNN operations
  - **Edge Cases**: Handle scenarios where k > available targets
  - **Attribute Preservation**: Verify additional columns are maintained in results
  - **Correctness Validation**: Cross-verify results against PostGIS using equivalent queries

**Benchmark Results**
  Performance comparison across three engines using both small (100 trips × 1000 buildings) and large (1000 trips × 2000
  buildings) datasets:

  ### Large Dataset Results
  | Engine   | k=1       | k=5       | k=10      |
  |----------|-----------|-----------|-----------|
  | SedonaDB | 3.32ms    | 6.50ms    | 9.38ms    |
  | DuckDB   | 211.19ms  | 209.19ms  | 202.81ms  |
  | PostGIS  | 2171.37ms | 2237.47ms | 2221.19ms |

  ### Small Dataset Results
  | Engine   | k=1      | k=5      | k=10     |
  |----------|----------|----------|----------|
  | SedonaDB | 0.93ms   | 1.16ms   | 1.49ms   |
  | DuckDB   | 11.52ms  | 11.71ms  | 12.21ms  |
  | PostGIS  | 161.93ms | 162.62ms | 165.21ms |

  **SedonaDB demonstrates 8-655× faster performance** than competitors across all scenarios.

### CODE REVIEW

File: benchmarks/test_knn.py [hunk 1]
```diff
@@ -89,138 +96,162 @@ def test_knn_performance(self, benchmark, k, use_spheroid, dataset_size):
         else:
             trip_table = "knn_trips_small"
             building_table = "knn_buildings"
-            trip_limit = 500
+            trip_limit = 1000
 
-        spheroid_str = "TRUE" if use_spheroid else "FALSE"
+        # Get the appropriate engine instance
+        eng = self._get_eng(engine)
```
[the remark this answers is not in the document]
Comment (zhangfengcdt) [reply benchmarks/test_knn.py:102]: The _get_eng method is defined in test_bench_base.py:86 and available to all benchmark classes.

**Comment by 2010YOUY01** [comment]
It seems even for the largest dataset (2000 buildings x 1000 trips), sedona is only using 1 CPU core for the `SpatialJoinExec` phase. I think if you try a larger dataset at the join probe (trips) side, the speedup relative to DuckDB/pg can be higher. (That's a really impressive work!)

I tried two data sizes:
2000 buildings x 1000 trips: 25ms
2000 buildings x 20000 trips: 50ms

## reproducer in `sedona-cli`:
```sql
-- Setup
CREATE OR REPLACE VIEW knn_buildings AS
SELECT
    geometry AS geom,
    ROUND(random() * 1000) AS building_id,
    'Building_' || CAST(ROUND(random() * 1000) AS VARCHAR) AS name
FROM sd_random_geometry('{
    "geom_type": "Polygon",
    "target_rows": 2000,
    "vertices_per_linestring_range": [4, 8],
    "size_range": [0.001, 0.01],
    "seed": 42
}');

CREATE OR REPLACE VIEW knn_trips AS
SELECT
    geometry AS geom,
    ROUND(random() * 100000) AS trip_id
FROM sd_random_geometry('{
    "geom_type": "Point",
    "target_rows": 100000,
    "seed": 43
}');



-- Benchmark Query
WITH trip_sample AS (
    SELECT trip_id, geom AS trip_geom
    FROM knn_trips
    LIMIT 20000                               -- <== This line is changed
),
building_with_geom AS (
    SELECT building_id, name, geom AS building_geom
    FROM knn_buildings
)
SELECT
    t.trip_id,
    b.building_id,
    b.name,
    ST_Distance(t.trip_geom, b.building_geom) AS distance
FROM trip_sample t
JOIN building_with_geom b ON ST_KNN(t.trip_geom, b.building_geom, 5, FALSE)
ORDER BY t.trip_id, distance;
```

## Reasons
The data source `sd_random_geometry` seems to produce batches in 1k size, and the execution layer is using the number of CPU cores as the parallelism degree by default. If there is only 1 batch, only one partition is working; my machine has 14 cores, so if I try 20000 as the probe side size, all partitions can be kept busy with a similar amount of work to do per partition as before.

It can be verified through `explain analyze verbose + query`, and see `output_rows` field inside `SpatialJoinExec`
